packages/maskbook/src/plugins/Trader/apis/trending/index.ts

```diff
@@ -283,9 +286,31 @@ export async function getCoinTrendingByKeyword(
     // prefer coins on the etherenum network
     const coin = coins.find((x) => x.eth_address) ?? first(coins)
     if (!coin) return null
+    if (dataProvider === DataProvider.UNISWAP) {
```

Comment:
Please move this logic to `getCoinTrendingById `.

Reply:
okay

Reply:
when dataProvider === DataProvider.UNISWAP, does not getCoinTrending 'ById'

Reply:
I move `getCoinTrendingFromUniswap` to apis/uniswap/index.